docs/reference/components/sinks/syslog.cue

```diff
@@ -0,0 +1,200 @@
+package metadata
+
+components: sinks: syslog: {
+	title: "Syslog"
+
+	classes: {
+		commonly_used: true
+		delivery:      "best_effort"
+		development:   "beta"
+		egress_method: "stream"
+		service_providers: []
+		stateful: false
+	}
+
+	features: {
+		buffer: enabled:      true
+		healthcheck: enabled: true
+		send: {
+			compression: enabled: false
+			encoding: enabled: false
+			send_buffer_bytes: {
+				enabled:       true
+				relevant_when: "mode = `tcp` or mode = `udp`"
+			}
+			keepalive: enabled: true
+			request: enabled:   false
+			tls: {
+				enabled:                true
+				can_enable:             true
+				can_verify_certificate: true
+				can_verify_hostname:    true
+				enabled_default:        false
+			}
+			to: {
+				service: services.syslog
+
+				interface: {
+					socket: {
+						api: {
+							title: "Syslog"
+							url:   urls.syslog
+						}
+						direction: "outgoing"
+						protocols: ["tcp", "udp", "unix"]
+						ssl: "required"
+					}
+				}
+			}
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	configuration:  sinks.socket.configuration & {
+	    "type": "type": string: enum: syslog: "The type of this component."
+		rfc5424: {
+			common: true
+			description: "If this is set to `true` message will be formatted according to the RFC5424 else the RFC3164 will be used."
+			required: false
+			warnings: []
+            type: bool: default: true
+		}
+		include_extra_fields: {
+			common: false
+			description: "If this is set to `true` all fields that have not been mapped to a Syslog field will be included as structured data in the resulting message."
+			required: false
+			warnings: []
+            type: bool: default: false
+		}
+		appname_key: {
```

Comment:
I'm curious what others think about these options. There are a few concerns I have:

1. We plan to support schemas in the future. If we know the schema, we can do this mapping automatically, and if we don't, we'll provide a manual fallback. In that case, these options would likely live under `encoding.schema.appname_key`, etc. I'm wondering if we want to set that precedence now or later?
2. There is an argument to be made that users should front this sink with a `remap` transform and map to static keys as defined by the `syslog` sink. In other words, dropping all of these options.

cc @jszwedko

Reply:
Yeah, we definitely want to improve this sort of "schema mapping" in Vector. Until we have defined a path forward there, I'm inclined to keep the status quo, which is `foo_key` fields like we have here, rather than introduce a change here that might not reflect the actual desired end state.

Reply:
We have another pattern in sinks of using templatable fields for this too. This would look something like:

```
appname = "{{ some_field }}" # default to `{{ app_name }}`
```